src/app/application/services/data-source/DataSourcePool.ts

```diff
@@ -0,0 +1,83 @@
+import CallableEventSource, { isClosed } from './CallableEventSource'
+import Router from './Router'
+import SharedPool from './SharedPool'
+
+export type DataSourceResponse<T> = {data: T | undefined, error: Error | undefined, refresh: () => void}
+export type Source = (params: Record<string, unknown>, source: {close: () => void}) => Promise<unknown>
+export type Sources = Record<string, Source>
+
+const get = (src: string, router: Router<Source>) => {
```

Comment:
I get `get` because is an HTTP `GET` but maybe this would be better as a `create` which would be symetric with `destroy`

Reply:
Missing return type btw and if you look at it `undefined` is impossible which makes a lot of `undefined` handling in SharedPool and DataSourcePool unnecessary

Reply:
I removed a lot of the `| undefined` elsewhere, this one doesn't seem to require a return type, but I added for readability anyway.

I also moved everything to use `CallableEventSource` rather than just `EventSource` (still later on I would like it to all use `EventSource` as a type)